src/domain/usecases/SendNotificationDataValuesUseCase.ts

```diff
@@ -0,0 +1,347 @@
+                    auditValue.dataElement.id === dv.dataElement &&
+                    auditValue.categoryOptionCombo.id === dv.categoryOptionCombo
+            );
+
+            const latestAuditValue = _(auditValuesByDe).first();
+            return {
+                ...dv,
+                oldValue: latestAuditValue?.value || "'  '",
+            };
+        });
+        return dataValuesWithOldValue;
+    }
+
+    private getDataSetConfig(dataSets: DataSet[], dataSetId: string, settings: Settings) {
+        const dataSet = dataSets.find(dataSet => dataSet.id === dataSetId);
+        if (!dataSet) {
+            throw Error(`Cannot find dataset: ${dataSetId}`);
+        }
+        return (
+            settings.dataSets[dataSet.id] ??
+            settings.dataSets[dataSet.code] ??
+            settings.dataSets[dataSet.name]
+        );
+    }
+
+    private async saveExecutionsWithEmail(
+        execution: NotificationDetail,
+        dataSetExecutions: NotificationStore,
+        key: string,
+        options: OptionsUseCase
+    ) {
+        const dateEmailSent = new Date().toISOString();
+        execution.lastDateSentEmail = dateEmailSent;
+        execution.lastUpdated = dateEmailSent;
+        log.debug(`Email sent ${dateEmailSent}`);
+
+        await this.saveExecutions(
+            {
+                ...dataSetExecutions,
+                [key]: {
+                    ...execution,
+                },
+            },
+            options
+        );
+    }
+
+    private async sendEmailToUsersInGroup(users: User[], subject: string, body: string) {
+        await promiseMap(users, async user => {
+            await this.notificationsRepository.send({
+                attachments: [],
+                subject,
+                body: {
+                    contents: body,
+                    type: "html",
+                },
+                recipients: [user.email],
+            });
+        });
+    }
+
+    private async getEmailTemplate(path: Path): Async<EmailTemplate> {
+        return jsonfile.readFile(path);
+    }
+
+    private async getUsersToNotify(dataValues: DataValue[], userGroup: string) {
+        const editorUsers = _(dataValues)
+            .map(dv => dv.storedBy)
+            .uniq()
+            .value();
+
+        const usersInGroup = await this.userRepository.getFromGroup([userGroup]);
+        return usersInGroup.filter(user => !editorUsers.includes(user.username));
+    }
+
+    private diffSinceLastUpdate(dataValues: DataValue[]) {
+        if (dataValues.length > 0) {
+            const lastDeUpdated = _(dataValues).maxBy(dv => dv.lastUpdated);
+            if (lastDeUpdated) {
+                const currentDate = new Date();
+                const diffInMinutes = this.getDifferenceInMinutes(
+                    new Date(lastDeUpdated.lastUpdated),
+                    currentDate
+                );
+                return diffInMinutes;
+            }
+        }
+        return -1;
```

Comment:
Magic values like `-1` are usually a red flag. Can we return `number | undefined`? (and change the caller accordingly)